client/history/client.go

```diff
@@ -1079,6 +1080,21 @@ func (c *clientImpl) GetFailoverInfo(
 	return c.client.GetFailoverInfo(ctx, request, append(opts, yarpc.WithShardKey(peer))...)
 }
 
+func (c *clientImpl) RatelimitUpdate(ctx context.Context, request *types.RatelimitUpdateRequest, opts ...yarpc.CallOption) (*types.RatelimitUpdateResponse, error) {
+	if len(opts) == 0 {
```

Comment:
how about making shard key an explicit parameter?

Reply:
I think this might end up being the nicest route... but doing that means changing quite a lot of code, as well as a few pieces of codegen :\

I'm happy to stick it on a "try it soon" to-do list (there are a few followups worth doing), but tbh it doesn't feel worth making this PR even bigger.  There's only one caller, and that's likely all there ever will be.